addons/mail/static/src/components/attachment/attachment_tests.js

```diff
@@ -595,6 +595,34 @@ QUnit.test('close attachment viewer', async function (assert) {
     );
 });
 
+QUnit.test('close attachment viewer before image load', async function (assert) {
+    assert.expect(1);
+
+    await this.start({ hasDialog: true });
+    const attachment = this.env.models['mail.attachment'].create({
+        filename: "test.png",
+        id: 750,
+        mimetype: 'image/png',
+        name: "test.png",
+    });
+    await this.createAttachmentComponent(attachment, {
+        detailsMode: 'hover',
+        isDownloadable: false,
+        isEditable: false,
+    });
+    await afterNextRender(() => document.querySelector('.o_Attachment_image').click());
+    const image = document.querySelector('.o_AttachmentViewer_viewImage');
+    await afterNextRender(() =>
+        document.querySelector('.o_AttachmentViewer_headerItemButtonClose').click()
+    );
+    image.dispatchEvent(new Event('load', { bubbles: true }));
+    assert.containsNone(
+        document.body,
+        '.o_Dialog',
+        "attachment viewer should be closed after clicking on close button"
+    );
+});
+
```

Comment:
It's short, sure, but the name doesn't give a spec. The purpose of test is rather vague.
Maybe this test crashed without fix, but this doesn't mean the test is good.

Can you tell exactly what feature didn't work without this fix? If it's the closing of dialog not working, then maybe the test itself is good and it's just the name is off. In that case, suggestion:

- Name: _"should close attachment viewer of image before image is loaded"_ (just a few additional words, and it makes purpose of test very explicit)
- Assert no attachment viewer rather than dialog. The dialog is just container

Reply:
Without the fix, there is a traceback on 'load' dispatch.

Reply:
Without the fix, there is a traceback (attachmentVIewer is undefined) when the 'load' event is dispatched, I changed for 0 assertions and a more explicit title.

Reply:
Hm... Don't really like this test, but considering the complexity with workaround using refs, it's fine to have a test.

Would add `[technical]` prefix on test name (with brief comment, it's good for `[technical]` tests to provide some assumption on how the code was designed at the moment of writing of test).

Rather than `assert.expect(0)`, I would prefer `assert.expect(1)` and use of `try/catch`. Adding assertion(s) where needed makes it super clear what the test is actually testing. Considering the test is about not crashing, it should be just `assert.ok()` at end of `try` or `assert.notOk` at end of `catch`.